README.md

```diff
@@ -20,7 +20,17 @@ Images
 Please refer to the [developer guide](https://www.osbuild.org/docs/developer-guide/index) to learn about our workflow, code style and more.
 
 The build-requirements for Fedora and rpm-based distributions are:
-- `gpgme-devel`, `btrfs-progs-devel`, `device-mapper-devel`
+- `btrfs-progs-devel`
+- `device-mapper-devel`
+- `gcc`
+- `git-core`
+- `go`
+- `gpgme-devel`
+- `gpgme-devel`
+- `krb5-devel`
+- `osbuild-depsolve-dnf`
```

Comment:
We might need to be more clear here, or link to a document explaining how things work, so that users and developers of the library understand what's going on.

This repository is, primarily, a Go library for generating osbuild manifests.  It also has some libraries for uploading stuff to cloud platforms and Koji.  All binaries defined in `cmd/` are for development and testing purposes (at least for now, ignoring otk externals).  Given all that, the only _real_ build and test dependencies are:
- `go`
- `gpgme-devel`

Everything else on that list is either a build or runtime dependency of a public package.

- `btrfs-progs-devel`, `device-mapper-devel`: build dependencies for projects that import `pkg/container`, which even in that case can be skipped using `exclude_graphdriver_btrfs` and `exclude_graphdriver_devicemapper` (see bootc-image-builder).
- `gcc`: outdated.  Used to be we needed it to compile a mock dnf-json binary, but that's long gone.
- `git-core`: not a dependency.  Needed to clone the repository, which is why you might have seen it in some test setup scripts.
- `krb5-devel`: build dependency for projects that import `pkg/upload/koji`.  Some unit tests for the package would be good and, when we add them, `krb5-devel` becomes a test dependency.
- `osbuild-depsolve-dnf`: runtime dependency for projects that import `pkg/dnfjson`.

Now if we also account for `cmd/build` and `cmd/gen-manifests`:
- `osbuild-depsolve-dnf`: runtime dependency for `cmd/gen-manifests` and `cmd/build`.
- `osbuild` (and subpackages): runtime dependencies for `cmd/build`.

Reply:
Thanks, obviously I was also lacking this knowledge
maybe can can align directly where a good place for this knowledge is

Reply:
Yes we should have this information somewhere in the repo.  If not in the README, then in a document linked from the README.

Reply:
I'll try to document in the README somehow (not yet committed)

Reply:
@achilleas-k please proof read what I understood.
I left `otk*` binaries aside on purpose and we should document them once settled (and not part of this PR)